### PR TITLE
[FLINK-22818][tests] Remembering important values only on certain che…

### PR DESCRIPTION
**Pull request by akalash**
…ckpoint

## What is the purpose of the change

*One more attempt to fix org.apache.flink.test.checkpointing.IgnoreInFlightDataITCase#testIgnoreInFlightDataDuringRecovery*

## Verifying this change

It is test itself

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-tests/src/test/java/org/apache/flink/test/checkpointing/IgnoreInFlightDataITCase.java [hunk 1]
```diff
@@ -255,7 +257,12 @@ public void cancel() {
...
         @Override
         public void snapshotState(FunctionSnapshotContext context) throws Exception {
-            if (lastCheckpointValue.get().get() > 0) {
+            if (context.getCheckpointId() > 2) {
+                // It is possible if checkpoint was triggered too fast after restart.
+                return; // Just ignore it.
+            }
+
+            if (context.getCheckpointId() == 2) {
                 throw new RuntimeException("Error during snapshot");
```
Comment (dawidwys) [inline flink-tests/src/test/java/org/apache/flink/test/checkpointing/IgnoreInFlightDataITCase.java:266]: nit: Could we adjust the message and maybe even use the `ExpectedTestException`? If I understand it correctly this is in most cases an expected failure, right?

**Review by dawidwys — APPROVED** [review]
After I understood the test, the fix makes sense. I had one nitpick, but otherwise, I think it's good. Let me know if you like to change the nitpick or should I merge it as is.

File: flink-tests/src/test/java/org/apache/flink/test/checkpointing/IgnoreInFlightDataITCase.java (hunk 1 above)
Comment (akalash) [reply flink-tests/src/test/java/org/apache/flink/test/checkpointing/IgnoreInFlightDataITCase.java]: not a problem. done

**Comment by akalash** [comment]
@dawidwys I changed the nitpick so you can merge it now(I think it doesn't make sense to wait for the test)

**Pull request merged**